File: src/Domino.Application/Models/GameView.cs
using Domino.Domain.Entities;

namespace Domino.Application.Models;

public class GameView
{
    public long Id { get; set; }
    public GameError? Error { get; set; }
    public Table Table { get; set; } = null!;
    public PlayerGameView Player { get; set; } = null!;
    public PlayerGameView Opponent { get; set; } = null!;
    public int MarketTilesCount { get; set; }
    public GameRules GameRules { get; set; } = null!;
    public GameStatus GameStatus { get; set; } = null!;
    public GameResult? GameResult { get; set; }
    public GameLog Log { get; set; } = null!;
}

File: src/Domino.UI/src/app/_models/gameView.ts
import { GameError } from "./gameError";
import { GameResult } from "./gameResult";
import { GameRules } from "./gameRules";
import { GameStatus } from "./gameStatus";
import { GameTable } from "./gameTable";
import { LogEvent } from "./logEvent"
import { PlayerGameView } from "./playerGameView";

export interface GameView {
    id: number;
    error: GameError;
    table: GameTable;
    player: PlayerGameView;
    opponent: PlayerGameView;
    marketTilesCount: number;
    gameStatus: GameStatus;
    gameResult?: GameResult;
    gameRules: GameRules;
    log: {
        events: LogEvent[]
    };
}